Code/Framework/AzToolsFramework/Tests/EditorViewportOnlyDrawHelpersForSelectedEntityTests.cpp

```diff
@@ -0,0 +1,186 @@
+#include <AzToolsFramework/ViewportSelection/EditorHelpers.h>
+
+namespace UnitTest
+{
+    class EditorViewportOnlyDrawHelpersForSelectedEntityFixture
+        : public ToolsApplicationFixture
+        , public AzFramework::EntityDebugDisplayEventBus::Handler
+    {
+    public:
+        inline static constexpr AzFramework::ViewportId TestViewportId = 2468;
+
+        void SetUpEditorFixtureImpl() override
+        {
+            AllocatorsTestFixture::SetUp();
+
+            // Setup entity to use for EntityDebugDisplayEventBus and tests
+            AZ::Entity* entity = nullptr;
+            m_entityId = CreateDefaultEditorEntity("ComponentModeEntity", &entity);
+
+            AzFramework::EntityDebugDisplayEventBus::Handler::BusConnect(m_entityId);
+
+            // DebugDisplay to use in DisplayHelpers
+            AzFramework::DebugDisplayRequestBus::BusPtr debugDisplayBus;
+            AzFramework::DebugDisplayRequestBus::Bind(debugDisplayBus, TestViewportId);
+            m_debugDisplay = AzFramework::DebugDisplayRequestBus::FindFirstHandler(debugDisplayBus);
+
+            m_cameraState = AzFramework::CreateDefaultCamera(AZ::Transform::CreateIdentity(), AzFramework::ScreenSize(1024, 768));
+
+            m_entityVisibleEntityDataCacheMock = AZStd::make_unique<::testing::NiceMock<MockEditorVisibleEntityDataCacheInterface>>();
+            m_editorHelpers = AZStd::make_unique<AzToolsFramework::EditorHelpers>(m_entityVisibleEntityDataCacheMock.get());
+            m_viewportSettings = AZStd::make_unique<ViewportSettingsTestImpl>();
+
+            m_viewportSettings->Connect(TestViewportId);
+            m_viewportSettings->m_helpersVisible = true;
+            m_viewportSettings->m_iconsVisible = true;
+
+            using ::testing::_;
+            using ::testing::Return;
+
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, GetVisibleEntityId(_)).WillByDefault(Return(AZ::EntityId(m_entityId)));
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, VisibleEntityDataCount()).WillByDefault(Return(1));
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, IsVisibleEntityIconHidden(_)).WillByDefault(Return(false));
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, IsVisibleEntityVisible(_)).WillByDefault(Return(true));
+            ON_CALL(*m_entityVisibleEntityDataCacheMock, IsVisibleEntitySelected(_))
+                .WillByDefault(
+                    [this](size_t)
+                    {
+                        AzToolsFramework::EntityIdList entityIds;
+                        AzToolsFramework::ToolsApplicationRequestBus::BroadcastResult(
+                            entityIds, &AzToolsFramework::ToolsApplicationRequests::GetSelectedEntities);
+
+                        if (AZStd::ranges::find(entityIds, m_entityId) != entityIds.end())
+                        {
+                            return true;
+                        }
+                        return false;
+                    });
+        }
```

Comment:
question: This looks good, but do we want all this to be in the fixture setup? Perhaps we only need this in one test? So either moving this to a function or moving to the test setup might be better? Not a huge deal but often it's useful to make test fixtures more reusable and not coupling them to one particular test if possible

Reply:
I want this to apply to every test because the DisplayHelpers function relies on these so if I moved them to the tests I have to call these every time?

Reply:
Okay if it's definitely needed for all it's fine to leave as-is, but it's also an option to move this to a function and then just call that function from the tests that need it (can help with flexibility at times but if we don't need that yet it's fine to stay as-is 👍